target/linux/ar71xx/files/arch/mips/ath79/mach-domywifi-dw33d.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * DomyWifi DW33D support
+ *
+ * Copyright (c) 2012 Qualcomm Atheros
+ * Copyright (c) 2012-2013 Gabor Juhos <juhosg@openwrt.org>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <linux/platform_device.h>
+#include <linux/ar8216_platform.h>
+#include <linux/mtd/mtd.h>
+#include <linux/mtd/nand.h>
+#include <linux/platform/ar934x_nfc.h>
+
+#include <asm/mach-ath79/ar71xx_regs.h>
+
+#include "common.h"
+#include "pci.h"
+#include "dev-ap9x-pci.h"
+#include "dev-gpio-buttons.h"
+#include "dev-eth.h"
+#include "dev-leds-gpio.h"
+#include "dev-m25p80.h"
+#include "dev-nfc.h"
+#include "dev-usb.h"
+#include "dev-wmac.h"
+#include "machtypes.h"
+
+#define DW33D_GPIO_LED_MMC		4
+#define DW33D_GPIO_LED_WLAN_2G		13
+#define DW33D_GPIO_LED_STATUS		14
+#define DW33D_GPIO_LED_USB		15
+#define DW33D_GPIO_LED_INTERNET		22
+
+#define DW33D_GPIO_BTN_RESET		17
+
+#define DW33D_KEYS_POLL_INTERVAL	20	/* msecs */
+#define DW33D_KEYS_DEBOUNCE_INTERVAL	(3 * DW33D_KEYS_POLL_INTERVAL)
+
+#define DW33D_MAC0_OFFSET		0
+#define DW33D_MAC1_OFFSET		6
+#define DW33D_WMAC_OFFSET		12
+#define DW33D_WMAC_CALDATA_OFFSET	0x1000
+#define DW33D_PCIE_CALDATA_OFFSET	0x5000
+
+static struct gpio_led dw33d_leds_gpio[] __initdata = {
+	{
+		.name		= "domy:blue:status",
+		.gpio		= DW33D_GPIO_LED_STATUS,
```

Comment:
leds should use the boardname. please use dw33d:blue:status. same for all of the leds